src/main/java/org/alliancegenome/curation_api/services/helpers/validators/NoteValidator.java

```diff
@@ -0,0 +1,125 @@
+package org.alliancegenome.curation_api.services.helpers.validators;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import org.alliancegenome.curation_api.dao.NoteDAO;
+import org.alliancegenome.curation_api.dao.VocabularyTermDAO;
+import org.alliancegenome.curation_api.exceptions.ApiErrorException;
+import org.alliancegenome.curation_api.model.entities.Note;
+import org.alliancegenome.curation_api.model.entities.VocabularyTerm;
+import org.alliancegenome.curation_api.response.*;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.*;
+
+@RequestScoped
+public class NoteValidator {
+
+    @Inject
+    NoteDAO noteDAO;
+    @Inject
+    VocabularyTermDAO vocabularyTermDAO;
+    
+    
+    
+    protected String invalidMessage = "Not a valid entry";
+    protected String obsoleteMessage = "Obsolete term specified";
+    protected String requiredMessage = "Required field is empty";
+    
+    protected ObjectResponse<Note> response;
+    
+    public Note validateNote(Note uiEntity, String noteVocabularyName, Boolean throwError) {
+        response = new ObjectResponse<>(uiEntity);
+        String errorTitle = "Could not update Note: [" + uiEntity.getId() + "]";
+        
+        Long id = uiEntity.getId();
+        if (id == null) {
+            addMessageResponse("No Note ID provided");
+            throw new ApiErrorException(response);
```

Comment:
@markquintontulloch you might want these to be `ObjectValidationException`? I can't remember how we did it for the other validators?

Reply:
@oblodgett - for everything else we've got ApiErrorExceptions for stuff coming in through the UI and ObjectValidationExceptions for the DTO validation.